src/libraries/System.Diagnostics.DiagnosticSource/ref/System.Diagnostics.DiagnosticSourceActivity.cs

```diff
@@ -146,6 +146,9 @@ public sealed class ActivitySource : IDisposable
         public string Name { get { throw null; } }
         public string? Version { get { throw null; } }
         public bool HasListeners() { throw null; }
+        public System.Diagnostics.Activity? CreateActivity(string name, System.Diagnostics.ActivityKind kind) { throw null; }
+        public System.Diagnostics.Activity? CreateActivity(string name, System.Diagnostics.ActivityKind kind, System.Diagnostics.ActivityContext parentContext, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, object?>>? tags = null, System.Collections.Generic.IEnumerable<System.Diagnostics.ActivityLink>? links = null) { throw null; }
+        public System.Diagnostics.Activity? CreateActivity(string name, System.Diagnostics.ActivityKind kind, string parentId, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, object?>>? tags = null, System.Collections.Generic.IEnumerable<System.Diagnostics.ActivityLink>? links = null) { throw null; }
```

Comment:
Weren't we going to handle #43853 by adding an IdFormat parameter here?

Reply:
No. After discussing this in the design review, @pakrym felt the other proposal is not needed and this can be handled by this CreateActivity. After creating the activity, they can set the IdFormat on such created Activity.

Reply:
Unforetunately that doesn't work. CreateActivity() calls the sampling callback, sampling callback has access to the ID, generating that ID has already fixed the id format. Using SetIdFormat() after CreateActivity() should probably throw an exception if it doesn't already.

Reply:
```C#
        public Activity SetIdFormat(ActivityIdFormat format)
        {
            if (_id != null || _spanId != null)
            {
                NotifyError(new InvalidOperationException(SR.SetFormatOnStartedActivity));
            }
            else
            {
                IdFormat = format;
            }
            return this;
        }
```


If the Activity not started, `_id` and `_spanId` would still be nulls. otherwise Activity.Start() would fail. no?

Reply:
@noahfalk I have addressed all comments. The remaining point is this ne which related to IdFormat. I don't think we should fix the Ids before starting the Activity. Please let me know if you think otherwise.

Reply:
I think we need to ensure that if we are creating a W3C activity then ActivityCreationOptions.TraceId that the sampler sees is the same trace Id that you get via Activity.TraceId. This implies it also shouldn't be legal to call SetIdFormat() after invoking the sampling callback because it would invalidate ActivityCreationOptions.TraceId that we already exposed.
It is probably easier to continue having _id and _spanId be computed lazily in Start() and that isn't a problem as long as we enforce the SetIdFormat() requirement some other way.

Reply:
>This implies it also shouldn't be legal to call SetIdFormat() after invoking the sampling callback because it would invalidate ActivityCreationOptions.TraceId that we already exposed.

This means we might need to bring the https://github.com/dotnet/runtime/issues/43853#issuecomment-780126996 feature request back.

Reply:
Let me clarify the current behavior:

Currently if anyone calling StartActivity (which will be same as CreateActivity), 

- When using parent ActivityContext:
    - If the caller passing none-default ActivityContext (which has the traceId), then we honor that trace Id and that will never change when starting the Activity. i.e. what the listener see as trace id will be same as Activity traceId.
    - If the caller passing default ActivityContext, if there is valid parent (Activity.Current), we'll us the context of the parent and the trace Id again will be same for what the listener see and what Activity trace Id return.
    - If the caller passing default ActivityContext, if there is NO valid parent (Activity.Current == null)
        - If the listeners try to access the trace Id from the ActivityCreationOption, we'll create a new context including trace Id. and that trace Id will be used for Activity.TraceId too.
        - If the listeners didn't access the trace Id, we'll generate trace Id when starting the activity. **This is the only time IdFormat can affect generating the trace Id.**

- When using parent Id (string and not a context)
    - If the Id is formatted as a way can be converted to Activity Context, the generated context will be used as described before.
    - If the Id cannot be converted, we'll generate a new trace Id at the Activity.Start(). **This is the only time IdFormat will affect generating the trace Id.**

So to summarize, even when providing IdFormat parameter to StartActivity, it will not affect anything except the cases:
- The caller is passing default context, there is no parent (Activity.Current), and the listeners didn't try to access the trace Id from the provided ActivityCreationOptions.
- The caller is passing parent id string instead of the context and this parent Id is not formatted to extract context from it.

Considering these 2 cases, it can be achieved by CreateActivity too. 

In short, adding IdFormat to StartActivity wouldn't give us more than what CreateActivity will. And setting IdFormat wouldn't affect anything except the 2 mentioned cases.

So, I am still we don't need #43853 anyway. And CreateActivity still enough.

Reply:
>The caller is passing default context, there is no parent (Activity.Current), and the listeners didn't try to access the trace Id from the provided ActivityCreationOptions.

This case seems very real. Think of a worker app uploading a blob. There is no parent and all spans are exported so TraceId is not accessed.

Reply:
> This case seems very real. Think of a worker app uploading a blob. There is no parent and all spans are exported so TraceId is not accessed.

Yes, agree. But using CreateActivity with the current design will make it work. right? I mean users can use CreateActivity and SetIdFormat if they need to for this scenario.

Reply:
@pakrym I had offline discussion with @noahfalk and we propose adding extra boolean parameter to the CreateActivity called `forceW3CId`. The reason we didn't have the parameter of IdFormat because we want to promote W3C in general and not allow users to try the hierarchal one. Please let's know if you have any concern. I'll send it anyway to the fxdc and will CC you there.

```
        public Activity? CreateActivity(
                             string name,
                             ActivityKind kind, 
                             ActivityContext parentContext,
                             IEnumerable<KeyValuePair<string, object?>>? tags = null, 
                             IEnumerable<ActivityLink>? links = null,
                             bool forceW3CId = false);

        public Activity? CreateActivity(
                             string name, 
                             ActivityKind kind,
                             string parentContext,
                             bool forceW3CId = false,
                             IEnumerable<KeyValuePair<string, object?>>? tags = null, 
                             IEnumerable<ActivityLink>? links = null,
                             bool forceW3CId = false);
```

Reply:
LGTM. If we want to promote W3C can we default the parameter to true? 😄

Reply:
> If we want to promote W3C can we default the parameter to true? 😄

I discussed that with @noahfalk and he preferred not to default the value to W3C as he want to have consistent behavior with StartActivity when passing the exact same parameters.